src/bzmesh.jl

```diff
@@ -45,17 +45,47 @@ function bzmesh_ir_wedge(kgrid_size, lattice, atoms; tol_symmetry=1e-5)
         "spglib failed to get the symmetries. Check your lattice, or use a uniform BZ mesh."
     )
 
-    # Checks: S is unitary (noting that S is given in lattice coordinates)
-    for S in eachslice(spg_symops["rotations"]; dims=1)
-        Scart = lattice * S * inv(lattice)  # Form S in cartesian coords
+    Stildes = spg_symops["rotations"]
+    τtildes = spg_symops["translations"]
+    n_symops = size(Stildes, 1)
+    @assert size(τtildes, 1) == n_symops
+    # Notice: In the language of the latex document in the docs
+    # spglib returns \tilde{S} and \tilde{τ} in integer real-space coordinates, such that
+    # (A Stilde A^{-1}) is the actual \tilde{S} from the document as a unitary matrix.
+    #
+    # Still we have the following properties for S and τ given is *integer* and
+    # *fractional* real-space coordinates:
+    #      - \tilde{S}^{-1} = S^T (if applied to a vector in frac. coords in reciprocal space)
+
+    # Checks: (A Stilde A^{-1}) is unitary
+    for Stilde in eachslice(Stildes; dims=1)
+        Scart = lattice * Stilde * inv(lattice)  # Form S in cartesian coords
         if maximum(abs, Scart'Scart - I) > sqrt(eps(Float64))
             error("spglib returned non-unitary rotation matrix")
         end
     end
-    # TODO check that S * lattice + τ ∈ lattice
+
+    # Check (Stilde, τtilde) maps atoms to equivalent atoms in the lattice
+    for (Stilde, τtilde) in zip(eachslice(Stildes; dims=1), eachslice(τtildes; dims=1))
+        for (elem, positions) in atoms
+            # Rationalize atomic positions
+            positions = [rationalize.(pos, tol=tol_symmetry) for pos in positions]
```

Comment:
Oh that's very cool

Reply:
Although that looks a bit brittle - are you sure it's not going to cause trouble, and wouldn't it be simpler to work in floats and just check the non-integer part is not too big?